Lib/smtplib.py

```diff
@@ -633,10 +634,13 @@ def auth(self, mechanism, authobject, *, initial_response_ok=True):
         if initial_response is not None:
             response = encode_base64(initial_response.encode('ascii'), eol='')
             (code, resp) = self.docmd("AUTH", mechanism + " " + response)
+            self._auth_challenge_count = 1
         else:
             (code, resp) = self.docmd("AUTH", mechanism)
+            self._auth_challenge_count = 0
         # If server responds with a challenge, send the response.
-        if code == 334:
+        while code == 334:
```

Comment:
This `while` statement is unguarded. What if server keeps sending the challenge, wont this code block the client indefinitely?

Reply:
I've implemented the guard in 481ca55

I think this conversation is now resolved.